### PR TITLE
Fix leveldb tests

### PR DESCRIPTION
**Pull request by aleksuss**
#280

### CODE REVIEW

File: 3rdparty/leveldb/tests/iterator.rs [hunk 1]
```diff
@@ -30,9 +33,9 @@ fn test_iterator_last() {
   db_put_simple(database, [2], &[2]);
 
   let read_opts = ReadOptions::new();
-  let iter = database.iter(read_opts);
+  let mut iter = database.iter(read_opts);
 
-  assert!(iter.last().is_some());
+  assert!(iter.next().is_some());
```
Comment (gisochre) [inline 3rdparty/leveldb/tests/iterator.rs]: this test is obsolete, is it? there's no last() in `Iterator` now

File: 3rdparty/leveldb/tests/iterator.rs [hunk 2]
```diff
@@ -45,13 +48,10 @@ fn test_iterator_from_to() {
   db_put_simple(database, [4], &[4]);
   db_put_simple(database, [5], &[5]);
 
-  let from = [2];
-  let to = [4];
   let read_opts = ReadOptions::new();
-  let mut iter = database.iter(read_opts).from(&from).to(&to);
```
Comment (gisochre) [inline 3rdparty/leveldb/tests/iterator.rs]: this test is obsolete too

File: 3rdparty/leveldb/tests/iterator.rs [hunk 3]
```diff
@@ -65,9 +65,9 @@ fn test_key_iterator() {
   let iterable: &mut Iterable = database;
 
   let read_opts = ReadOptions::new();
-  let mut iter = iterable.keys_iter(read_opts);
```
Comment (gisochre) [inline 3rdparty/leveldb/tests/iterator.rs]: obsolete

File: 3rdparty/leveldb/tests/iterator.rs [hunk 4]
```diff
@@ -80,7 +80,7 @@ fn test_value_iterator() {
   let iterable: &mut Iterable = database;
 
   let read_opts = ReadOptions::new();
-  let mut iter = iterable.value_iter(read_opts);
-  let value = iter.next().unwrap();
-  assert_eq!(value, vec![1]);
+  let mut iter = iterable.iter(read_opts);
```
Comment (gisochre) [inline 3rdparty/leveldb/tests/iterator.rs]: obsolete

File: 3rdparty/leveldb/tests/iterator.rs (hunk 1 above)
Comment (aleksuss) [inline 3rdparty/leveldb/tests/iterator.rs]: That's right. This modifications was made only for compiling and running without errors

File: 3rdparty/leveldb/tests/iterator.rs (hunk 2 above)
Comment (aleksuss) [inline 3rdparty/leveldb/tests/iterator.rs]: +

File: 3rdparty/leveldb/tests/iterator.rs (hunk 3 above)
Comment (aleksuss) [inline 3rdparty/leveldb/tests/iterator.rs]: `+`

File: 3rdparty/leveldb/tests/iterator.rs (hunk 4 above)
Comment (aleksuss) [inline 3rdparty/leveldb/tests/iterator.rs]: `+`

**Comment by gisochre** [comment]
@aleksuss removing them or let them stay?

**Comment by aleksuss** [comment]
I see nothing bad in their present. Travis build doesn't depend on they.

**Comment by gisochre** [comment]
i'd remove them tho, why maintain dead api

**Comment by aleksuss** [comment]
@gisochre I guess in a few month we'll remove leveldb at all from core with this dead tests

**Pull request merged**